src/ucp/rma/amo_sw.c

```diff
@@ -46,18 +46,18 @@ static size_t ucp_amo_sw_pack(void *dest, void *arg, int fetch, int new_proto)
     return length;
 }
 
-static size_t ucp_amo_sw_post_pack_cb(void *dest, void *arg)
+static size_t ucp_amo_sw_post_pack_cb(void *dest, ucp_request_t *req)
 {
-    return ucp_amo_sw_pack(dest, arg, 0, 0);
+    return ucp_amo_sw_pack(dest, req, 0, req->send.length);
 }
 
-static size_t ucp_amo_sw_fetch_pack_cb(void *dest, void *arg)
+static size_t ucp_amo_sw_fetch_pack_cb(void *dest, ucp_request_t *req)
```

Comment:
maybe revert? then would not need to cast
there is not requirement that offload pack cb would be same as sw pack cb